### PR TITLE
Add hostname field to Boot Register request

### PR DESCRIPTION
**Pull request by GrigoriyMikhalkin**
cc @robertvolkmann 

Added `hostname` field to Boot gRPC service Register request. The logic handles it in such way that the field is expected only from `metal-hammer` at the moment. With possibility of extending functionality of this field in a future.

### CODE REVIEW

File: cmd/metal-api/internal/metal/network.go [hunk 1]
```diff
@@ -166,3 +167,13 @@ func (nics Nics) ByName() map[string]*Nic {
 	}
 	return res
 }
+
+func (nics Nics) FilterByHostname(hostname string) (res Nics) {
+	for i, n := range nics {
```
Comment (majst01) [inline cmd/metal-api/internal/metal/network.go:172]: For backward compatibility we should return all nics if hostname is empty string

WDYT

Comment (Gerrit91) [reply cmd/metal-api/internal/metal/network.go:172]: Also looks breaking to me. The switch nics do not have a hostname set at the moment, how is this supposed to find any nics for existing environments?

Comment (GrigoriyMikhalkin) [reply cmd/metal-api/internal/metal/network.go:172]: @Gerrit91 I think we already store hostnames for switch NICs. The current version of metal-hammer doesn't send the hostname -- but omitting hostname filter for NICs should work as current switches will be identified by MAC.

Comment (Gerrit91) [reply cmd/metal-api/internal/metal/network.go:176]: Looked it up again. 
- Switches do not have hostnames stored in their nics
- However, you do not look up the hostname in a switch nic but only for machine nic  
- The `ConnectMachine` function is only called during register, so these changes should be fine with an updated metal-hammer version

**Pull request closed without merging**